Test_openjdk11_j9_extended.functional_x86-64_linux (+xl) threadMXBeanTestSuite1 SAXParseException

https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_extended.functional_x86-64_linux_xl_Nightly/94
```
===============================================
Running test threadMXBeanTestSuite1_5 ...
===============================================
threadMXBeanTestSuite1_5 Start Time: Wed Jul 17 06:12:03 2019 Epoch Time (ms): 1563358323550
variation: Mode301 -Xjit:noJitUntilMain,count=0,optlevel=warm
JVM_OPTIONS: -Xnocompressedrefs -Xgcpolicy:metronome -Xnocompressedrefs -Xjit:noJitUntilMain,count=0,optlevel=warm 
[TestNG] [ERROR] org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 8; The processing instruction must begin with the name of the target.
Exception in thread "main" java.lang.NullPointerException
	at org.testng.TestNG.getStatus(TestNG.java:211)
	at org.testng.TestNG.main(TestNG.java:1324)

threadMXBeanTestSuite1_5_FAILED
```

This occurrence happened before the one above.
https://ci.eclipse.org/openj9/job/Test_openjdk13_j9_extended.functional_x86-64_linux_OpenJDK13/15
threadMXBeanTestSuite1_6
```
===============================================
Running test threadMXBeanTestSuite1_6 ...
===============================================
threadMXBeanTestSuite1_6 Start Time: Tue Jul 16 02:00:52 2019 Epoch Time (ms): 1563253252620
variation: Mode351 -Xjit:noJitUntilMain,count=0,optlevel=warm
JVM_OPTIONS: -Xcompressedrefs -Xgcpolicy:metronome -Xcompressedrefs -Xjit:noJitUntilMain,count=0,optlevel=warm 
[TestNG] [ERROR] org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 8; The processing instruction must begin with the name of the target.
Exception in thread "main" java.lang.NullPointerException
	at org.testng.TestNG.getStatus(TestNG.java:211)
	at org.testng.TestNG.main(TestNG.java:1324)

threadMXBeanTestSuite1_6_FAILED
```

Tried the latest testng, the test still fail, but there's no NPE this time. The real failure is
```
[TestNG] [ERROR] org.xml.sax.SAXParseException; lineNumber: 1; columnNumber: 8; The processing instruction must begin with the name of the target.
```

--- a/runtime/compiler/x/codegen/J9TreeEvaluator.cpp
+++ b/runtime/compiler/x/codegen/J9TreeEvaluator.cpp
@@ -3012,11 +3012,35 @@ TR::Register *J9::X86::TreeEvaluator::BNDCHKwithSpineCHKEvaluator(TR::Node *node
             }
          else
             {
-            // Check the bounds.
-            //
-            TR::TreeEvaluator::compareIntegersForOrder(node, indexChild, arrayLengthChild, cg);
-            branchOpCode = JAE4;
-            faultingInstruction = cg->getImplicitExceptionPoint();
+            TR::DataType dt = loadOrStoreChild->getDataType();
+            int32_t elementSize = (dt == TR::Address) ?  TR::Compiler->om.sizeofReferenceField()
+                                                      : TR::Symbol::convertTypeToSize(dt);
+
+            if (TR::Compiler->om.isDiscontiguousArray(arrayLengthChild->getInt(), elementSize))
+               {
+               // Create real check failure snippet if we can prove the spine check
+               // will always fail
+               //
+               branchOpCode = JMP4;
+               cg->decReferenceCount(arrayLengthChild);
+               if (!indexChild->getOpCode().isLoadConst())
+                  {
+                  cg->evaluate(indexChild);
+                  }
+               else
+                  {
+                  cg->decReferenceCount(indexChild);
+                  }
+               faultingInstruction = cg->getImplicitExceptionPoint();
+               }
+            else
+               {
+               // Check the bounds.
+               //
+               TR::TreeEvaluator::compareIntegersForOrder(node, indexChild, arrayLengthChild, cg);
+               branchOpCode = JAE4;
+               faultingInstruction = cg->getImplicitExceptionPoint();
+               }
             }
          }
       else

